lib/builder/builder.js

```diff
@@ -273,7 +273,8 @@ module.exports = class Builder {
 
     // -- Layouts --
     if (existsSync(resolve(this.options.srcDir, this.options.dir.layouts))) {
-      const layoutsFiles = await glob(`${this.options.dir.layouts}/**/*.{vue,js}`, {
+      const extensions = this.options.pageExtensions
```

Comment:
What if we extend/simplify usage of [options.extensions](https://github.com/nuxt/nuxt.js/blob/dev/lib/common/options.js#L191) for pages? 

/cc @Atinux @clarkdo

Reply:
We need to add an example of differentiating what happens to an `.md` file when it's loaded as a page and in a page/js module.

Also, not everything that can be a page can be a middleware?
Are we sure there are no downsides to checking all extensions in all locations?

Reply:
I have changed the example from md to mjs, as seems is confusing... Nuxt doesn't need to be aware of what the page content is or what extension is using as long as it finally gets what it expects, a Vue Component.

Reply:
Having a separated option for pages seems reasonable. However, still, miss leading as it also covers layouts 😆

Reply:
Agree, I was expecting some feedback here. Should it be pages specific or general (in which case we need another option name).

I vote for general as eventually the same features on pages will be expected in layouts and others.

Reply:
Currently, these directories will be auto-scanned.

- Type A: Accept **components** only including `js`, `ts`, `mjs`, `vue`, `md`, etc
  - Configurable with ??
- Type B: Accept any `js`,`mjs`, `ts` file and exports are **not** components.
  - Configurable with `options.extensions`

- pages : A
- layouts : A
- store (js/mjs/ts): B
- middleware: B

Reply:
Is there any prior work on importing template languages as vue components?

Technically, something like
```vue
<template lang="md" src="./boop.md"></template>
```
```vue
<template lang="pug" src="./boop.pug"></template>
```
could be a generic solution.

On the other hand, maybe it should be a `<div v-once v-html="the stuff">` to allow multiple root nodes.

Reply:
I believe that file extensions are for us, humans, so we can easily recognize a file and easily apply some filter, loaders, etc. to those. But bots don't care as they only care about file type.

Maybe I am missing something here and expecting linux behaviour where it cannot, but if doable I would just keep `options.extensions` as global supported extensions and process all the files from pages/layouts/store... the same way.

Reply:
@qm3ster, the official nuxt markdown module add support for md language on vue templates, if that was your question about.

Reply:
Nah, I meant loading as vue component constructor, not language for template.